src/main/java/de/hpi/bpt/argos/api/eventQuery/EventQueryEndpointImpl.java

```diff
@@ -1,44 +1,213 @@
+
+		return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String deleteEventQuery(Request request, Response response) {
+        long eventQueryId = getEventQueryId(request);
+        EventQuery eventQuery = PersistenceAdapterImpl.getInstance().getEventQuery(eventQueryId);
+        if (eventQuery == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "given query was not found");
+        }
+
+		EventType queryOwner = PersistenceAdapterImpl.getInstance().getEventType(eventQuery.getTypeId());
+        if (queryOwner == null) {
+        	halt(HttpStatusCodes.ERROR, "cannot find query owner");
+		}
+
+		if (!queryOwner.isDeletable()
+				&& PersistenceAdapterImpl.getInstance().getEventQueries(queryOwner.getId()).size() == 1) {
+			halt(HttpStatusCodes.FORBIDDEN, "this query must not be deleted");
+		}
+
+        EventPlatformFeedback feedback = EventProcessingPlatformUpdaterImpl.getInstance().deleteEventQuery(eventQuery);
+        if (!feedback.isSuccessful()) {
+            halt(HttpStatusCodes.ERROR, String.format("cannot unregister event query: %1$s", feedback.getResponseText()));
+        }
+
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifact(eventQuery,
+                EventTypeEndpoint.getEventTypeQueriesUri(eventQuery.getId()))) {
+            halt(HttpStatusCodes.ERROR, "could not delete query");
+        }
+
+        return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String editEventQuery(Request request, Response response) {
+        JsonObject jsonEventQuery = new JsonObject();
+        try {
+            jsonEventQuery = jsonParser.parse(request.body()).getAsJsonObject();
+        } catch (JsonSyntaxException e) {
+            LoggerUtilImpl.getInstance().error(logger, JSON_PARSE_ERROR_MESSAGE, e);
+            halt(HttpStatusCodes.BAD_REQUEST, "not a valid json");
+        }
+
+        if (jsonEventQuery == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "no event query given in body");
+        }
+
+        EventQuery newEventQuery = getEventQueryFromJson(jsonEventQuery, false);
+        if (newEventQuery == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "failed to parse event query");
+        }
+
+        long oldEventQueryId = getEventQueryId(request);
+        EventQuery oldEventQuery = PersistenceAdapterImpl.getInstance().getEventQuery(oldEventQueryId);
+        if (oldEventQuery == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "given query was not found");
+        }
+
+        EventPlatformFeedback deleteFeedback = EventProcessingPlatformUpdaterImpl.getInstance().deleteEventQuery(oldEventQuery);
+        if (!deleteFeedback.isSuccessful()) {
+            halt(HttpStatusCodes.ERROR, "given query could not be unregistered");
+        }
+
+        oldEventQuery.setDescription(newEventQuery.getDescription());

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'oldEventQuery' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'newEventQuery' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)